### ISSUE TITLE
feat(CI/CD):Implement Code Cov for CI checks

### ISSUE THREAD

**Issue by lsampras** [juspay/hyperswitch#1593]
Tracked in #1587 
Implement Code coverage for PR's via CI

Objectives
- run tests as part of CI
- generate codeCov for these tests & auto-publish the report as part of PR Comment...
- [Optionally] be able to specify that atleast x% of new code should be test-covered (the CI test fails if this is not met)

**Comment by carrycooldude** [comment]
I will love to work on this issue @lsampras , can you assign me this and Can you elobrate more on this?

**Comment by lsampras** [comment]
Hey @carrycooldude ,
this is already tracked (& assigned) in #1587....

You can check the above issue for related issues

**Comment by tarunsamanta2k20** [comment]
@carrycooldude please go ahead. I am facing issues with hardware.

**Comment by lsampras** [comment]
Hey @carrycooldude ,
I've added you as the assignee for this issue,

You can ask your doubts on this issue or our discord/slack...

**Comment by aialok** [comment]
Is this issue still available. I am ready to work on this. I already setup Codecov in one of the open source project.
Thank you : )

**Comment by SanchithHegde** [comment]
@aialok Yes, this issue is open for contribution. Do let us know if you'd like us to assign this issue to you.

**Comment by aialok** [comment]
Yes. I would love to work on this issue.
Please assign : )

**Comment by SanchithHegde** [comment]
@aialok Before you start working on this, hold on, let's check if @carrycooldude is still working on their PR #1700 or if we can close it. Apologies for the confusion.

**Comment by aialok** [comment]
> @aialok Before you start working on this, hold on, let's check if @carrycooldude is still working on their PR #1700 or if we can close it. Apologies for the confusion.

Alright, please assign this to me if @carrycooldude is not working. I believe we should set up something that automatically labels stale issues using https://github.com/actions/stale.
